src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,45 @@
-import React from 'react';
+import { UserInfo } from '../UserInfo';
+import commentsFromServer from '../../api/comments';
+import usersFromServer from '../../api/users';
+import { Post } from '../../types/post';
+import { CommentList } from '../CommentList';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+type Props = {
+  post: Post,
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const {
+    userId,
+    title,
+    body,
+    id,
+  } = post;
+
+  const user = usersFromServer.find(curUser => curUser.id === userId);
+  const comments = commentsFromServer
+    .filter(curComment => curComment.postId === id);
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{title}</h3>
+
+        <p>
+          {' Posted by  '}
+
+          {user && <UserInfo dataType="post" user={user} />}
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      <hr />
+      {comments.length === 0
+        ? <b data-cy="NoCommentsMessage">No comments yet</b>
+        : <CommentList comments={comments} />}
+    </div>
```

Comment:
tag `hr` has to be only when there is no comments added, maybe that is why some tests had failed

Reply:
nope) this position of hr is perfect

Reply:
> tag `hr` has to be only when there is no comments added, maybe that is why some tests had failed

nope) this position of hr is perfect

Reply:
tried both variations, problem is not here)